packages/eslint-plugin/src/rules/no-useless-template-literals.ts

```diff
@@ -54,7 +54,10 @@ export default createRule<[], MessageId>({
     }
 
     function isLiteral(expression: TSESTree.Expression): boolean {
-      return expression.type === AST_NODE_TYPES.Literal;
+      return (
+        expression.type === AST_NODE_TYPES.Literal ||
+        expression.type === AST_NODE_TYPES.TemplateLiteral
```

Comment:
```suggestion
     return (
        expression.type === AST_NODE_TYPES.Literal ||
        (expression.type === AST_NODE_TYPES.TemplateLiteral &&
          expression.expressions.length === 0 &&
          expression.quasis.length === 1)
```

I think we should also check for the existence of expressions in the template literal.
maybe, 


Because the auto-fixed result will cause a parsing error if an expression is present

- [playground preview](https://deploy-preview-8575--typescript-eslint.netlify.app/play/#ts=5.3.3&fileType=.tsx&code=MYewdgzgLgBGCuBbGBeGBGA3AKAAYDMQQASAb1wCMBDAJzIUQF9dHqAvXTIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUSWWS18iALbF4AQ2G1Gw6BPioMkRNGgdokcGAC%2BIHUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

- srouce
```ts
const num = 1;
`foo${`bar${num}`}baz`;
```
- after fixed
```ts
const num = 1;
`foo`bar${num}`baz`; // Parsing Error
```

Reply:
Thanks for pointing out. I'll update and add this test case.

Reply:
And there seems to be one more thing to think about when an expression exists.

```ts
const variable = '...';

`foo${`bar${variable}bar}foo`;
```
This code contains useless template literal also it can be reported and fixed to below.

```ts
const variable = '...';
`foobar${variable}barfoo`;
```

but I'm not sure this case should be handled in this PR, I'd love to hear from maintainers.

Reply:
Added a [valid case](https://github.com/typescript-eslint/typescript-eslint/pull/8575/files#diff-e1397d81c6335b2044a2b15a62abf713bd821075d1a7b1d1f37521f031b1acefR66) about this situation, which can be fixed later if planned.

Reply:
IMO, it's better to just file a follow-up issue rather than adding a possibly invalid (in future) test as a valid test. So we can clarify if this is intended behavior or not (personally I'm +1 on having this). Are you up for filing this as a new issue?

Reply:
> it's better to just file a follow-up issue rather than adding a possibly invalid (in future) test as a valid test

Yes, that makes sense to me. I'll open a new issue to keep track of this.